### PR TITLE
grpcproxy: add filter to watcher

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: proxy/grpcproxy/watcher.go [hunk 1]
```diff
@@ -37,13 +39,32 @@ func (w *watcher) send(wr clientv3.WatchResponse) {
 		return
 	}
 
-	// todo: filter out the events that this watcher already seen.
```
Comment (heyitsanthony) [inline proxy/grpcproxy/watcher.go]: but this doesn't do that?

Comment (xiang90) [inline proxy/grpcproxy/watcher.go]: removed wrong todo. will fix.

File: proxy/grpcproxy/watch.go [hunk 2]
```diff
@@ -104,6 +105,7 @@ func (sws *serverWatchStream) recvLoop() error {
 				ch: sws.watchCh,
 
 				progress: cr.ProgressNotify,
+				filters:  v3rpc.FiltersFromRequest(cr),
```
Comment (heyitsanthony) [inline proxy/grpcproxy/watch.go]: can an unfiltered watcher on the same range be coalesced into a filtered watcher?

Comment (xiang90) [inline proxy/grpcproxy/watch.go]: This is for the watcher inside the proxy. The watcher we created for etcd server is always unfiltered for better coalesce purpose. see line 116.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**